src/SerializeAST.cc

```diff
@@ -0,0 +1,634 @@
+//
+//  SerializeAST.cc
+//  snowcrash
+//
+//  Created by Pavan Kumar Sunkara on 18/01/15.
+//  Copyright (c) 2015 Apiary Inc. All rights reserved.
+//
+
+#include "SerializeAST.h"
+
+using namespace snowcrash;
+
+sos::Object wrapValue(const mson::Value& value)
+{
+    sos::Object valueObject;
+
+    // Literal
+    valueObject.set(SerializeKey::Literal.c_str(), sos::String(value.literal));
+
+    // Variable
+    valueObject.set(SerializeKey::Variable.c_str(), sos::Boolean(value.variable));
+
+    return valueObject;
+}
+
+sos::Object wrapSymbol(const mson::Symbol& symbol)
+{
+    sos::Object symbolObject;
+
+    // Literal
+    symbolObject.set(SerializeKey::Literal.c_str(), sos::String(symbol.literal));
+
+    // Variable
+    symbolObject.set(SerializeKey::Variable.c_str(), sos::Boolean(symbol.variable));
+
+    return symbolObject;
+}
+
+sos::Base wrapTypeName(const mson::TypeName& typeName)
+{
+    if (typeName.base != mson::UndefinedTypeName) {
+
+        std::string baseTypeName = "";
+
+        switch (typeName.base) {
+
+            case mson::BooleanTypeName:
+                baseTypeName = "boolean";
+                break;
+
+            case mson::StringTypeName:
+                baseTypeName = "string";
+                break;
+
+            case mson::NumberTypeName:
+                baseTypeName = "number";
+                break;
+
+            case mson::ArrayTypeName:
+                baseTypeName = "array";
+                break;
+
+            case mson::EnumTypeName:
+                baseTypeName = "enum";
+                break;
+
+            case mson::ObjectTypeName:
+                baseTypeName = "object";
+                break;
+
+            default:
+                break;
+        }
+
+        return sos::String(baseTypeName);
```

Comment:
It also strikes me that you are not constructing it as sos::String directly. This really points to the odd concept of this pseudo-generic sos library. I hope @klokane will eventually tighten this up.